packages/authentication/src/__tests__/acceptance/authentication-action-provider/authentication-action.provider.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: loopback4-example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Getter, inject, Provider, Setter} from '@loopback/context';
```

Comment:
You can rewrite the [provider](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/providers/authentication.provider.ts) directly.

See [the same file in my PR](https://github.com/strongloop/loopback-next/blob/53cfeebbaf2ebe2897e28f51432f6fb5d1a4e946/packages/authentication/src/providers/authentication.provider.ts), which contains exactly the same code as yours, and also proves this action is agnostic of the strategy type(passport or non-passport).